src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -708,6 +708,12 @@ private static void ScanLdtoken (Instruction operation, Stack<StackSlot> current
 				return;
 			}
 
+			if (operation.Operand is ArrayType arrayType) {
+				StackSlot slot = new StackSlot (new ArrayValue (new ConstIntValue (arrayType.Rank)));
```

Comment:
This doesn't look right. It essentially makes `typeof(Foo[])` act like `new Foo[1]`.

We'll need to grab the TypeDefinition of System.Array.

Can you add a test that will fail for the code that is here now?

You can unit test that `typeof(int[]).GetProperty("LongLength")` is a recognized pattern. I don't think the recognition will work now because the expected result of ScanLdtoken is a `RuntimeTypeHandleValue`, not `ArrayValue`.